### PR TITLE
Export flac properly when use with add_subdirectory

### PR DESCRIPTION
**Pull request by martinRenou**
Mimicking **https://github.com/xiph/vorbis/commit/98eddc72d36e3421519d54b101c09b57e4d4d10d**

This fixes issues when using `add_subdirectory` with libflac

### CODE REVIEW

**Comment by ktmf01** [comment]
Could you explain this a little more perhaps? I don't fully grasp the concept. These changes are necessary when FLAC is used as a direct dependency of another project, more or less? Especially the `if(NOT TARGET Ogg::ogg)`, is that if FLAC is build as a subdirectory of ogg?

In other words: how is this used? Currently the CMake CI process on Windows doesn't build with ogg, if building FLAC and OGG together is made easier, this would make building ogg for CI easier as well

**Comment by martinRenou** [comment]
> These changes are necessary when FLAC is used as a direct dependency of another project, more or less? 

Yes

> Especially the if(NOT TARGET Ogg::ogg), is that if FLAC is build as a subdirectory of ogg?

Not really, this is in case Ogg was already built in the same build tree. For example:

```cmake

set(OGG_GIT_TAG         v1.3.5)
# master contains fixes for creating a build tree
# https://github.com/xiph/vorbis/commit/98eddc72d36e3421519d54b101c09b57e4d4d10d
# set(VORBIS_GIT_TAG      v1.3.7)
set(VORBIS_GIT_TAG      master)
# Latest Flac with https://github.com/xiph/flac/pull/285
# set(FLAC_GIT_TAG        1.3.3)
set(FLAC_GIT_TAG        export_flac)

# ogg
# ---

message(STATUS "Building ogg")

set(INSTALL_DOCS OFF CACHE BOOL "" FORCE)

FetchContent_Declare(
  ogg
  GIT_REPOSITORY https://github.com/xiph/ogg
  GIT_TAG ${OGG_GIT_TAG}
)

FetchContent_GetProperties(ogg)

if(NOT ogg_POPULATED)
  FetchContent_Populate(ogg)
  add_subdirectory(${ogg_SOURCE_DIR} ${ogg_BINARY_DIR})
endif()

# VORBIS
# ------

message(STATUS "Building libvorbis")

set(BUILD_TESTING OFF CACHE BOOL "" FORCE)

FetchContent_Declare(
  vorbis
  GIT_REPOSITORY https://github.com/xiph/vorbis
  GIT_TAG ${VORBIS_GIT_TAG}
)

FetchContent_GetProperties(vorbis)

if(NOT vorbis_POPULATED)
  FetchContent_Populate(vorbis)
  add_subdirectory(${vorbis_SOURCE_DIR} ${vorbis_BINARY_DIR})
endif()

# FLAC
# ----

message(STATUS "Building libflac")

set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
set(BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
set(BUILD_DOXYGEN OFF CACHE BOOL "" FORCE)
set(BUILD_DOCS OFF CACHE BOOL "" FORCE)
set(INSTALL_MANPAGES OFF CACHE BOOL "" FORCE)

FetchContent_Declare(
  FLAC
  # GIT_REPOSITORY https://github.com/xiph/flac
  GIT_REPOSITORY https://github.com/martinRenou/flac
  GIT_TAG ${FLAC_GIT_TAG}
)

FetchContent_GetProperties(FLAC)

if(NOT FLAC_POPULATED)
  FetchContent_Populate(FLAC)
  add_subdirectory(${flac_SOURCE_DIR} ${flac_BINARY_DIR})
endif()
```

Without the fix that this PR suggests, the call to `add_subdirectory(${flac_SOURCE_DIR} ${flac_BINARY_DIR})` would fail.

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,9 @@ option(WITH_OGG "ogg support (default: test for libogg)" ON)
 set(VERSION ${PROJECT_VERSION})
 
 if(WITH_OGG)
-    find_package(Ogg REQUIRED)
+    if(NOT TARGET Ogg::ogg)
+        find_package(Ogg REQUIRED)
+    endif()
```
Comment (ktmf01) [inline CMakeLists.txt:30]:
```suggestion
    else()
        set(OGG_FOUND 1 CACHE INTERNAL "ogg has already been built")
    endif()
```

**Review by ktmf01 — CHANGES_REQUESTED** [review]
I just tested the script you presented, and with this, FLAC doesn't include ogg support.

```
[ 52%] Building C object _deps/flac-build/src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_intrin_sse2.c.o
[ 53%] Building C object _deps/flac-build/src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_intrin_ssse3.c.o
[ 54%] Building C object _deps/flac-build/src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_intrin_avx2.c.o
[ 55%] Building C object _deps/flac-build/src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_framing.c.o
[ 56%] Building C object _deps/flac-build/src/libFLAC/CMakeFiles/FLAC.dir/window.c.o
[ 57%] Linking C static library libFLAC.a
```

When building FLAC on its own, I get
```
[ 21%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_intrin_sse2.c.o
[ 22%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_intrin_ssse3.c.o
[ 22%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_intrin_avx2.c.o
[ 23%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/stream_encoder_framing.c.o
[ 24%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/window.c.o
[ 25%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/ogg_decoder_aspect.c.o
[ 26%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/ogg_encoder_aspect.c.o
[ 26%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/ogg_helper.c.o
[ 27%] Building C object src/libFLAC/CMakeFiles/FLAC.dir/ogg_mapping.c.o
[ 28%] Linking C static library libFLAC.a
```
I just added a change that seems to fix this. Can you verify this works correctly?

**Comment by ktmf01** [comment]
@martinRenou Could you take another look at this?

**Comment by martinRenou** [comment]
Thanks for pinging me again. I commited your suggestion, there seems to be a conflict I'll try to find time to resolve it in the coming days.

**Comment by ktmf01** [comment]
I just rebased your changes to solve the conflict. The conflict originated from PR #325 which touches the same bit of CMakeLists.txt.

Can you check whether this rebased version of your patch still works as expected before I merge it?

**Review by martinRenou — COMMENTED** [review]
LGTM! Thanks!

**Pull request merged**